From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sat, 15 Oct 2022 18:46:20 +0800
Subject: [PATCH] Fix incorrect cast in Blocks.java


diff --git a/src/main/java/net/minecraft/world/level/block/Blocks.java b/src/main/java/net/minecraft/world/level/block/Blocks.java
index 92c02fd5dd6e6941cce18ce5abe9d07f6df31445..8f965eb0358f962cd66d56dafc5537ace2758c0e 100644
--- a/src/main/java/net/minecraft/world/level/block/Blocks.java
+++ b/src/main/java/net/minecraft/world/level/block/Blocks.java
@@ -1091,7 +1091,7 @@ public class Blocks {
     }
 
     private static Boolean ocelotOrParrot(BlockState state, BlockGetter world, BlockPos pos, EntityType<?> type) {
-        return (boolean)type == EntityType.OCELOT || type == EntityType.PARROT;
+        return type == EntityType.OCELOT || type == EntityType.PARROT; // PlanetPaper - fix incorrect cast that make we cant compile.
     }
 
     private static BedBlock bed(DyeColor color) {
