From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sat, 15 Oct 2022 20:53:48 +0800
Subject: [PATCH] fixup! Optimize `Math.round` and `Math.hypot` functions


diff --git a/src/main/java/carpetfixes/helpers/FastMath.java b/src/main/java/carpetfixes/helpers/FastMath.java
index 2f48a1c71356f7ebc00aaa44b28f6e10b2de1dee..1563b21eac06ba0cc57336a639a2d83accb511c9 100644
--- a/src/main/java/carpetfixes/helpers/FastMath.java
+++ b/src/main/java/carpetfixes/helpers/FastMath.java
@@ -10,8 +10,7 @@ public class FastMath {
      * ~1.25x faster than {@link Math#round(float)}
      */
     public static int round(float a) {
-        if (wtf.etil.mirai.MiraiConfig.optimizedRound) return a > 0F ? (int)(a + .5F) : (int)(a - .5F);
-        return Math.round(a);
+        return a > 0F ? (int)(a + .5F) : (int)(a - .5F);
     }
 
     /**
@@ -19,8 +18,7 @@ public class FastMath {
      * ~1.28x faster than {@link Math#round(double)}
      */
     public static long round(double a) {
-        if (wtf.etil.mirai.MiraiConfig.optimizedRound) return a > 0D ? (long)(a + .5D) : (long)(a - .5D);
-        return Math.round(a);
+        return a > 0D ? (long)(a + .5D) : (long)(a - .5D);
     }
 
     /**
diff --git a/src/main/java/net/minecraft/util/Mth.java b/src/main/java/net/minecraft/util/Mth.java
index 41fe8bbdf9235ce79bffe6b2fe1339f79f34822f..0afd56a8b27fce044f7d43b1e4f86b617c0b827f 100644
--- a/src/main/java/net/minecraft/util/Mth.java
+++ b/src/main/java/net/minecraft/util/Mth.java
@@ -792,8 +792,7 @@ public class Mth {
     }
 
     public static double length(double a, double b) {
-        if (wtf.etil.mirai.MiraiConfig.optimizedHypot) return carpetfixes.helpers.FastMath.hypot(a, b); // Mirai
-        return Math.sqrt(lengthSquared(a, b));
+        return carpetfixes.helpers.FastMath.hypot(a, b); // Mirai
     }
 
     public static double lengthSquared(double a, double b, double c) {
