From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nostalgic853 <yuu8583@proton.me>
Date: Sun, 16 Oct 2022 14:41:19 +0800
Subject: [PATCH] Add snakeyaml dependency


diff --git a/build.gradle.kts b/build.gradle.kts
index 902f06aaf5d1d8bc9addd10991cf1c7fda80b444..4b53cc11321d2eb52db02e2e687d72c4278bed69 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -55,6 +55,13 @@ dependencies {
 
     implementation("io.netty:netty-all:4.1.77.Final"); // Paper - Bump netty
     implementation(kotlin("stdlib")) // PlanetPaper
+
+    // PlanetPaper start - snakeyaml.
+    implementation("org.yaml:snakeyaml:1.32")
+    implementation ("me.carleslc.Simple-YAML:Simple-Yaml:1.8.2") {
+        exclude(group="org.yaml", module="snakeyaml")
+    }
+    // PlanetPaper end
 }
 
 val craftbukkitPackageVersion = "1_19_R1" // Paper
